src/backend/renderer.js

```diff
@@ -746,19 +781,22 @@ export function attach(
     }
     const id = getFiberID(primaryFiber);
     if (isRoot) {
-      beginNextOperation(2);
+      // Removing a root needs to happen at the end
+      // so we don't batch it with other unmounts.
+      beginNextOperation(3);
       nextOperation[0] = TREE_OPERATION_REMOVE;
-      nextOperation[1] = id;
-      endNextOperation(false);
+      nextOperation[1] = 1; // Remove one item
+      nextOperation[2] = id;
+      endNextOperation();
     } else if (!shouldFilterFiber(fiber)) {
-      beginNextOperation(2);
-      nextOperation[0] = TREE_OPERATION_REMOVE;
-      nextOperation[1] = id;
-      // Non-root fibers are deleted during the commit phase.
-      // They are deleted in the parent-first order. However
-      // DevTools currently expects deletions to be child-first.
-      // This is why we prepend the delete operation to the queue.
-      endNextOperation(true);
+      // To maintain child-first ordering,
+      // we'll push it into one of these queues,
+      // and later arrange them in the correct order.
+      if (isSimulated) {
+        pendingSimulatedUnmountedIDs.push(id);
+      } else {
+        pendingRealUnmountedIDs.push(id);
+      }
```

Comment:
Before, this added each entry to the queue in reverse order (via `splice`). Now it no longer reverses the order (since it uses `push`). I'm guessing this is okay or it would fail tests and trigger invariants, but it surprises me. Am I misreading something?

Reply:
Maybe you’re missing the part where we’re sending the operations? The React-reported unmounts are added in the reverse order to typed array.

Reply:
Ah! I missed the fact that you were flipping the order in the other `for` loop. Gotcha!  👍